lte/gateway/c/session_manager/StatsPoller.cpp

```diff
@@ -13,17 +13,31 @@
 #include <stdint.h>
 #include <atomic>
 #include <memory>
+#include <vector>
 #include "StatsPoller.h"
-#define COOKIE 0
-#define COOKIE_MASK 0
+#define OVS_COOKIE_MATCH_ALL 0xffffffff
 
 namespace magma {
 
 void StatsPoller::start_loop(
     std::shared_ptr<magma::LocalEnforcer> local_enforcer,
-    uint32_t loop_interval_seconds) {
+    uint32_t loop_interval_seconds,
+    std::shared_ptr<magma::ShardTracker> shard_tracker) {
+  // when no shard ids are available poll all UEs
+  // to poll by cookie pass in a vector of shard ids, where
+  // each element is a shard id that maps directly to a cookie
   while (true) {
-    local_enforcer->poll_stats_enforcer(COOKIE, COOKIE_MASK);
+    // TODO(veshkemburu): add smart polling function to determine which
+    // shard ids to poll
+    if (shard_tracker->get_shard_list_size() == 0) {
```

Comment:
Is this ever going to be true? I saw this in the constructor
```

ShardTracker::ShardTracker() {
  // initialize with at least one empty vector to avoid checking for empty
  imsis_per_shard_.push_back({});
}
```